include/seqan3/io/sam_file/detail/format_sam_base.hpp

```diff
@@ -238,32 +208,51 @@ inline void format_sam_base::read_forward_range_field(stream_view_type && stream
     }
 }
 
+/*!\brief Reads from `str` to `target`, converting values with seqan3::views::char_to.
+ * \tparam target_range_type The type of range to parse from input; must model std::ranges::forward_range.
+ *
+ * \param[in, out] str          The string view to parse.
+ * \param[out]     target       The range to store the parsed sequence.
+ */
+template <std::ranges::forward_range target_range_type>
+inline void format_sam_base::read_forward_range_field(std::string_view const str, target_range_type & target)
+{
+    if (!(str.size() == 1 && str[0] == '*')) // '*' denote empty field
+    {
+        if constexpr (std::constructible_from<target_range_type, std::string_view>)
+        {
+            target = str;
+        }
+        else
+        {
+            target.resize(str.size());
+            for (size_t i = 0; i < str.size(); ++i)
+                target[i] = assign_char_to(str[i], std::ranges::range_value_t<target_range_type>{});
```

Comment:
```suggestion
            target.reserve(str.size());
            for (auto && chr : str)
                target.emplace_back(assign_char_to(chr, std::ranges::range_value_t<target_range_type>{}));
```

Would be an alternative.

But since we only require forward_range, we cannot rely on `operator[]`, and probably resize/reserve/emplace_back?